service/debugger/debugger.go

```diff
@@ -701,6 +716,13 @@ func (d *Debugger) AmendBreakpoint(amend *api.Breakpoint) error {
 	defer d.targetMutex.Unlock()
 
 	originals := d.findBreakpoint(amend.ID)
+	if originals == nil {
+		if _, ok := d.disabledBreakpoints[amend.ID]; ok {
+			d.disabledBreakpoints[amend.ID] = amend
```

Comment:
Here you should check if `amend.Disabled` is false and if it is re-enable the breakpoint, like you do in the first half of `(*Debugger).ToggleBreakpoint`.

After the ValidBreakpointName check you should check if `amend.Disabled` is true and if it is disable the breakpoint.

This way the functionality of ToggleBreakpoint can be provided through the `AmendBreakpoint` API interface.